### ISSUE TITLE
BUG/ERR: DatetimeIndex - Period shows ununderstandable error

### ISSUE THREAD

**Issue by sinhrks** [pydata/pandas#13078]
#### Code Sample, a copy-pastable example if possible

Split from #13071. ``DatetimeIndex`` - ``Period`` is not supported ATM, but errors are not understandable.

```
pd.DatetimeIndex(['2011-01-01', '2011-02-01']) - pd.Period('2011-01-01', freq='D') 
# ValueError: Cannot do arithmetic with non-conforming periods

pd.DatetimeIndex(['2011-01-01', '2011-01-02'], freq='D') - pd.Period('2011-01-01', freq='D') 
# AttributeError: 'DatetimeIndex' object has no attribute 'ordinal'
```

#### Expected Output

show understandable error message. 

Or it should be supported converting ``Period`` to ``Timestamp``.

```
pd.DatetimeIndex(['2011-01-01', '2011-02-01']) - pd.Period('2011-01-01', freq='D').to_timestamp()
# TimedeltaIndex(['0 days', '31 days'], dtype='timedelta64[ns]', freq=None)
```

#### output of ``pd.show_versions()``

on current master

**Comment by MaximilianR** [comment]
>show understandable error message.

I vote for this. 
I can't think of a reasonable way there could be arithmetic between points-in-time and spans. Tenuously, we _could_ convert a Period into a Timedelta and use that, but then the user could do that explicitly

**Comment by jreback** [comment]
I agree an error is best here - maybe show how to convert in the error message though 

dfi and period arithmetic cannot be intermixed implicitly

**Comment by sinhrks** [comment]
This has been fixed by #13071, adding tests and release note)
